packages/zero-client/src/client/custom.ts

```diff
@@ -117,7 +117,12 @@ export class TransactionImpl<TSchema extends Schema = DefaultSchema>
 {
   readonly location = 'client';
   readonly mutate: MutateCRUD<TSchema>;
-  readonly query: SchemaQuery<TSchema>;
+  /**
+   * @deprecated Use {@linkcode createBuilder} with `tx.run(zql.table.where(...))` instead.
+   */
+  readonly query: TSchema['enableLegacyQueries'] extends true
+    ? SchemaQuery<TSchema>
+    : {};
```

Comment:
Why not `undefined`?

That is what I used for `mutateBatch`:

https://github.com/rocicorp/mono/blob/26d0cc0c1df7e43cf7cd36cdcc841f135ada866a/packages/zero-client/src/client/crud.ts#L31-L34

Reply:
Maybe just change SchemaQuery so we do not need to remember to do this in all the callers.